File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
const maxArea = function(height) {
    let result = 0;
    let right = height.length - 1;
    let left = 0;
    
    while(left < right){
        let small = Math.min(height[left], height[right]);
        let area = (right - left) * small;
        
        if(area > result) result = area;
        
        if(height[left] < height[right]) left++;
        else
            right--;
    }
    return result;
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int maxi = 0;
        int right = height.length - 1;

        while(left < right){
            if(height[left] < height[right]){
                maxi = Math.max(maxi, height[left] * (right - left));
                left++;
            }else{
                maxi = Math.max(maxi, height[right] * (right - left));
                right--;
            }
        }
        return maxi;
    }
}